internal/jet/string_expression.go

```diff
@@ -16,6 +16,9 @@ type StringExpression interface {
 	BETWEEN(min, max StringExpression) BoolExpression
 	NOT_BETWEEN(min, max StringExpression) BoolExpression
 
+	ANY_EQ(rhs ArrayExpression[StringExpression]) BoolExpression
+	ALL_EQ(rhs ArrayExpression[StringExpression]) BoolExpression
+
```

Comment:
If we add `ANY_EQ`, we would also need to add, for each of the types, NOT_EQ_ANY, LT_ANY, LT_EQ_ANY, etc... This would lead to number of methods explosion.

Simpler way would be to have `ANY` and `ALL` as generic functions:

```golang
func ANY[E Expression](array Array[E]) E {
    // check rangeTypeCaster
}
```

Then we can write for any type:

```golang
SomeStringColumn.EQ(ANY(Table.StringArrayColumn))
```

Reply:
Yeah, that makes sense. I'll change that

Reply:
Done